src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/ReferenceConfidenceModel.java

```diff
@@ -0,0 +1,547 @@
+        } else {
+            debuggingWriter = null;
+        }
+    }
+
+    /**
+     * Get the VCF header lines to include when emitting reference confidence values via calculateRefConfidence
+     * @return a non-null set of VCFHeaderLines
+     */
+    public Set<VCFHeaderLine> getVCFHeaderLines() {
+        final Set<VCFHeaderLine> headerLines = new LinkedHashSet<>();
+        headerLines.add(new VCFSimpleHeaderLine(GATKVCFConstants.SYMBOLIC_ALLELE_DEFINITION_HEADER_TAG, GATKVCFConstants.NON_REF_SYMBOLIC_ALLELE_NAME, "Represents any possible alternative allele at this location"));
+        return headerLines;
+    }
+
+    /**
+     * Close down this reference model, closing down any debugging information opened during execution
+     */
+    public void close() {
+        if ( debuggingWriter != null ) debuggingWriter.close();
+    }
+
+    /**
+     * Calculate the reference confidence for a single sample given the its read data
+     *
+     * Returns a list of variant contexts, one for each position in the {@code activeRegion.getLoc()}, each containing
+     * detailed information about the certainty that the sample is hom-ref for each base in the region.
+     *
+     *
+     *
+     * @param refHaplotype the reference haplotype, used to get the reference bases across activeRegion.getLoc()
+     * @param calledHaplotypes a list of haplotypes that segregate in this region, for realignment of the reads in the
+     *                         readLikelihoods, corresponding to each reads best haplotype.  Must contain the refHaplotype.
+     * @param paddedReferenceLoc the location of refHaplotype (which might be larger than activeRegion.getLoc())
+     * @param activeRegion the active region we want to get the reference confidence over
+     * @param readLikelihoods a map from a single sample to its PerReadAlleleLikelihoodMap for each haplotype in calledHaplotypes
+     * @param ploidyModel indicate the ploidy of each sample in {@code stratifiedReadMap}.
+     * @param model genotyping model.
+     * @param variantCalls calls made in this region.  The return result will contain any variant call in this list in the
+     *                     correct order by genomic position, and any variant in this list will stop us emitting a ref confidence
+     *                     under any position it covers (for snps and insertions that is 1 bp, but for deletions its the entire ref span)
+     * @return an ordered list of variant contexts that spans activeRegion.getLoc() and includes both reference confidence
+     *         contexts as well as calls from variantCalls if any were provided
+     */
+    public List<VariantContext> calculateRefConfidence(final Haplotype refHaplotype,
+                                                       final Collection<Haplotype> calledHaplotypes,
+                                                       final GenomeLoc paddedReferenceLoc,
```

Comment:
Please migrate from `GenomeLoc` to `SimpleInterval` everywhere in this branch -- the work of porting is not complete if this is not done.